### PR TITLE
connect Login component to /login endpoint

### PR DESCRIPTION
**Pull request by Hadeel-Salamin**
- [x] add props.title to Gbutton shared component so u can change the label.

- [x] add props.onClick to Gbutton shared component so u can pass the function you want to be done once the user click on it.

- [x] change Gbutton label with props.title in GetDetails component.

- [x] save inputs in state

- [x] call /login endpoint to check if username and passwords matches and if yes move to itemsList component "page".

### CODE REVIEW

**Review by FidaaHersh — APPROVED** [review]

**Review by amanshawar — APPROVED** [review]

File: client/package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "react-dom": "^16.8.4",
     "react-router-dom": "^5.0.0",
     "react-scripts": "2.1.8",
-    "styled-components": "^4.1.3"
+    "styled-components": "^4.1.3",
+    "@babel/runtime": ""
```
Comment (jema28) [inline client/package.json:12]: What is this doing?

File: client/src/components/Login/index.js [hunk 2]
```diff
@@ -14,38 +14,67 @@ import {
...
   }
 
+  login = () => {
+    const { history } = this.props;
+    const inputs = { username: this.state.username, password: this.state.password };
+    axios.post('/login', inputs).then(({ data }) => {
+      if (data.success) {
+        history.push('/items-page');
+      } else {
+        history.push('/error');
```
Comment (jema28) [inline client/src/components/Login/index.js:35]: Does this exist already?

File: client/src/components/Shared/GreenButton/index.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 import React from 'react';
 import StyledButton from './greenbutton.style';
 
-const GButton = () => (<StyledButton type="button">CONTINUE</StyledButton>
+const GButton = props => (
+  <StyledButton onClick={props.onClick} type="button">
+    {props.title}
```
Comment (jema28) [inline client/src/components/Shared/GreenButton/index.js:6]: You could destructure here but not necessary.

**Review by jema28 — CHANGES_REQUESTED** [review]
Half review done, will go through it again after the conflicts have been solved.

File: client/src/components/Login/index.js (hunk 2 above)
Comment (Hadeel-Salamin) [reply client/src/components/Login/index.js:35]: no

File: client/package.json (hunk 1 above)
Comment (orjwan98) [reply client/package.json:12]: Same question as @jema28

**Review by orjwan98 — APPROVED** [review]

File: client/src/components/Shared/GreenButton/index.js (hunk 3 above)
Comment (Hadeel-Salamin) [reply client/src/components/Shared/GreenButton/index.js:6]: > You could destructure here but not necessary.

No you can't?

File: client/package.json (hunk 1 above)
Comment (Hadeel-Salamin) [reply client/package.json:12]: I got error for @babel/runtime so I had to add this to avoid the error

**Review by jema28 — APPROVED** [review]

**Pull request merged**